src/app/mapControls/Filter.js

```diff
@@ -0,0 +1,227 @@
+define([
+    'app/config',
+    'app/mapControls/_CollapsePanel',
+
+    'dijit/_TemplatedMixin',
+    'dijit/_WidgetBase',
+    'dijit/_WidgetsInTemplateMixin',
+
+    'dojo/_base/declare',
+    'dojo/_base/lang',
+    'dojo/dom-class',
+    'dojo/dom-construct',
+    'dojo/on',
+    'dojo/query',
+    'dojo/text!app/mapControls/templates/Filter.html',
+
+    'bootstrap-stylus/js/button',
+    'bootstrap-stylus/js/collapse',
+    'bootstrap-stylus/js/tooltip',
+    'bootstrap-stylus/js/transition'
+], function (
+    config,
+    _CollapsePanel,
+
+    _TemplatedMixin,
+    _WidgetBase,
+    _WidgetsInTemplateMixin,
+
+    declare,
+    lang,
+    domClass,
+    domConstruct,
+    on,
+    query,
+    template
+) {
+    var c = declare([_WidgetBase, _TemplatedMixin, _WidgetsInTemplateMixin, _CollapsePanel], {
+        // description:
+        //      A control for filtering by a defined set of choices.
+        //      Allows selection of one or more choices.
+
+
+        templateString: template,
+        baseClass: 'filter panel panel-default',
+
+        // selectedValues: String[]
+        //      The currently selected items
+        selectedValues: null,
+
+        // any: Boolean
+        //      any or all
+        any: true,
+
+
+        // properties passed in via the constructor
+
+        // items: [String, String][]
+        //      description, value pairs
+        items: null,
+
+        // fieldName: String
+        //      The name of the field associated with this filter
+        fieldName: null,
+
+        // fieldType: String (see TYPE* constants below)
+        //      The type of the field so that we can build a proper query
+        fieldType: null,
+
+        // anyAllToggle: Boolean (default: false)
+        //      If true the any/all toggle buttons appear
+        anyAllToggle: false,
+
+        // relatedTableQuery: Boolean (default: false)
+        //      If true this is a query on related tables
+        relatedTableQuery: false,
+
+        constructor: function () {
+            // summary:
+            //      apply base class
+            console.log('app/mapControls/Filter:constructor', arguments);
+
+            this.selectedValues = [];
+        },
+        postCreate: function () {
+            // summary:
+            //      build bubbles
+            console.log('app/mapControls/Filter:postCreate', arguments);
+
+            this.items.forEach(function (item) {
+                var lbl = domConstruct.create('label', {
+                    'class': 'btn btn-default btn-xs',
+                    innerHTML: ' ' + item[0]
+                }, this.buttonContainer);
+                domConstruct.create('input', {
+                    value: item[1],
+                    type: 'checkbox',
+                    autocomplete: 'off',
+                    'onclick': lang.partial(lang.hitch(this, 'itemClicked'), item[1])
```

Comment:
You are right that was broken. I had to change the markup for the latest upgrade in bootstrap and didn't notice that I broke it. 

I decided to move the `onclick` up to the label instead of using event delegation because it was easier to get the field value as `item[1]` instead of trying to get it from the hidden checkbox element from the click event object on the label.